addons/website/models/ir_ui_view.py

```diff
@@ -270,6 +270,15 @@ def get_inheriting_views_arch(self, model):
         # prefer inactive website-specific views over active generic ones
         return views.filter_duplicate().filtered('active')
 
+    @api.model
+    def combine_with_specific_views(self, views, ids_to_check):
+        """This method add some specific view that do not have XML ID
+        """
+        if self._context.get('website_id'):
+            keys = self.browse(views).mapped('key')
+            views += self.browse(ids_to_check).filtered_domain([('website_id', '!=', False), ('key', 'in', keys)]).ids
```

Comment:
ids_to_check are ordered by `ORDER BY priority, id;` maybe this should be kept by doing instead something like:

```py
views = self.browse(ids_to_check).filtered_domain([
    '|', '&', ('website_id', '!=', False), ('key', 'in', keys),  ('id', 'in', views)
]).ids
```